File: 1-TwoSum/1-TwoSum.py
# Last updated: 8/20/2026, 2:20:48 AM
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        hm = {}

        for i in range(len(nums)):
            diff = target - nums[i]

            if diff in hm:
                return [i, hm[diff]]

            hm[nums[i]] = i

        return []
        

File: 1-TwoSum/1-TwoSum.ts
// Last updated: 8/20/2026, 2:20:50 AM
function twoSum(nums: number[], target: number): number[] {
    
    let hm = new Map();
    let result = [];
    
    for(let i=0; i <= nums.length; i++){
        if(hm.has(target - nums[i])){
            return [hm.get(target - nums[i]), i]
        }
        
        else{
            hm.set(nums[i], i)
        }
            
    };

};